# HG changeset patch
# User Rik <rik@octave.org>
# Date 1392604950 28800
#      Sun Feb 16 18:42:30 2014 -0800
# Branch gui-release
# Node ID 00d684465379f622e4421300df430a078e1be35d
# Parent  d39fd728ddef1488c1314a56b61241b86c7144da
axis.m: Make "image" and "square" compatible with Matlab (bug #41581).

* axis.m: Change plotboxaspectratiomode to "auto" when doing "image" format.
Set dataaspectratiomode to "auto" when doing "square" format.

diff --git a/scripts/plot/appearance/axis.m b/scripts/plot/appearance/axis.m
--- a/scripts/plot/appearance/axis.m
+++ b/scripts/plot/appearance/axis.m
@@ -180,19 +180,21 @@ function limits = __axis__ (ca, ax, vara
     if (strcmpi (ax, "ij"))
       set (ca, "ydir", "reverse");
     elseif (strcmpi (ax, "xy"))
       set (ca, "ydir", "normal");
 
       ## aspect ratio
     elseif (strcmpi (ax, "image"))
       __axis__ (ca, "equal");
+      set (ca, "plotboxaspectratiomode", "auto");
       __do_tight_option__ (ca);
     elseif (strcmpi (ax, "square"))
-      set (ca, "plotboxaspectratio", [1, 1, 1]);
+      set (ca, "dataaspectratiomode", "auto",
+               "plotboxaspectratio", [1, 1, 1]);
     elseif (strcmp (ax, "equal"))
       if (strcmp (get (get (ca, "parent"), "__graphics_toolkit__"), "gnuplot"))
         ## FIXME - gnuplot applies the aspect ratio activepostionproperty.
         set (ca, "activepositionproperty", "position");
         ## The following line is a trick used to trigger the recalculation of
         ## aspect related magnitudes even if the aspect ratio is the same
         ## (useful with the x11 gnuplot terminal after a window resize)
         set (ca, "dataaspectratiomode", "auto");
